ui/pages/send/gas-display/gas-display.js

```diff
@@ -0,0 +1,290 @@
+                      target="_blank"
+                      rel="noopener noreferrer"
+                    >
+                      {t('transactionDetailGasTooltipConversion')}
+                    </a>
+                  </p>
+                </>
+              }
+              position="top"
+            >
+              <i className="fa fa-info-circle" />
+            </InfoTooltip>
+          </>
+        }
+        detailText={
+          <Box
+            height={BLOCK_SIZES.MAX}
+            display={DISPLAY.FLEX}
+            flexDirection={FLEX_DIRECTION.COLUMN}
+            className="gas-display__currency-container"
+          >
+            {renderHeartBeatIfNotInTest()}
+            <UserPreferencedCurrencyDisplay
+              type={SECONDARY}
+              value={hexMinimumTransactionFee}
+              hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
+            />
+          </Box>
+        }
+        detailTotal={
+          <Box
+            height={BLOCK_SIZES.MAX}
+            display={DISPLAY.FLEX}
+            flexDirection={FLEX_DIRECTION.COLUMN}
+            className="gas-display__currency-container"
+          >
+            {renderHeartBeatIfNotInTest()}
+            <UserPreferencedCurrencyDisplay
+              type={PRIMARY}
+              value={hexMinimumTransactionFee}
+              hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+              numberOfDecimals={6}
+            />
+          </Box>
+        }
+        subText={
+          <>
+            <strong key="editGasSubTextFeeLabel">
+              {t('editGasSubTextFeeLabel')}
+            </strong>
+            <Box
+              height={BLOCK_SIZES.MAX}
+              display={DISPLAY.FLEX}
+              flexDirection={FLEX_DIRECTION.COLUMN}
+              className="gas-display__currency-container"
+            >
+              {renderHeartBeatIfNotInTest()}
+              <UserPreferencedCurrencyDisplay
+                key="editGasSubTextFeeAmount"
+                type={PRIMARY}
+                value={hexMaximumTransactionFee}
+                hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+              />
+            </Box>
+          </>
+        }
+        subTitle={
+          <>
+            <GasTiming
+              maxPriorityFeePerGas={hexWEIToDecGWEI(
+                maxPriorityFeePerGas ||
+                  draftTransaction.gas.maxPriorityFeePerGas,
+              )}
+              maxFeePerGas={hexWEIToDecGWEI(
+                maxFeePerGas || draftTransaction.gas.maxFeePerGas,
+              )}
+            />
+          </>
+        }
+      />
+    );
+  };
+  return (
+    <>
+      {showDepositPopover && (
+        <DepositPopover onClose={() => setShowDepositPopover(false)} />
+      )}
+      <Box className="gas-display">
+        <TransactionDetail
+          userAcknowledgedGasMissing={false}
+          rows={[
+            renderGasDetailsItem(),
```

Comment:
Either make this a new component, or put the JSX inline. I realize its a lot of JSX but if you want to break it out you should use components not functions. for example have a gas-details.js file that does this portion and then render it instead of calling a function.

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/c64948a3c0494308f824cdc443a9b04b010b9054).